### PR TITLE
kernel/task/task_prctl.c : Change to get the TCB associated with the …

### PR DESCRIPTION
**Pull request by jeongchanKim**
…PID for all cases

In previous PR_GET/SET_NAME, for pid 0, it gets the current running tcb.
In this case, we cannot get the Idle task's tcb.
So change to always get based on the pid.

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

### CODE REVIEW

File: os/kernel/task/task_prctl.c [hunk 1]
```diff
@@ -133,15 +133,9 @@ int prctl(int option, ...)
...
-		 * pid==0 meaning "this thread")
-		 */
+		/* Get the TCB associated with the PID */
 
-		if (!pid) {
-			tcb = this_task();
-		} else {
-			tcb = sched_gettcb(pid);
-		}
+		tcb = sched_gettcb(pid);
```
Comment (jeongarmy) [inline os/kernel/task/task_prctl.c:138]: I think we'd better keep original to get/set name of calling thread at once without pid by calling getpid().

Comment (sunghan-chang) [reply os/kernel/task/task_prctl.c:138]: Then, how about using `MAX_PID_MASK`? If arg3 is MAX_PID_MASK, then it uses `this_task` to get current running task name.

Comment (jeongchanKim) [reply os/kernel/task/task_prctl.c:138]: @jeongarmy , As I mentioned in the commit msg, it is better to use its own pid.
@sunghan-chang , using `MAX_PID_MASK` is not good for understanding, I think.

**Review by jeongarmy — APPROVED** [review]
Looks good

**Review by sunghan-chang — COMMENTED** [review]
Could you add and modify os/include/sys/prctl.h for description?

File: os/kernel/task/task_prctl.c [hunk 2]
```diff
@@ -133,14 +135,12 @@ int prctl(int option, ...)
...
 		FAR struct tcb_s *tcb;
 
-		/* Get the TCB associated with the PID (handling the special case of
-		 * pid==0 meaning "this thread")
-		 */
-
-		if (!pid) {
-			tcb = this_task();
-		} else {
+		if (option == PR_SET_NAME_BYPID || option == PR_GET_NAME_BYPID) {
```
Comment (sunghan-chang) [inline os/kernel/task/task_prctl.c:138]: Need to add `(option == PR_SET_NAME_BYPID)` in line 166

Comment (jeongchanKim) [reply os/kernel/task/task_prctl.c:138]: Added. Thanks.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**